Adding Internet stuff. WebSocket is now apart of CNE.

ItsLJcool opened the pull request.

ItsLJcool committed:
Added WebSocketUtil (and intenet folder)

Planning on adding more stuff to it.

ItsLJcool committed:
WebSocketUtil document

ItsLJcool committed:
Added the WebSockets and made `internet` lib

Hi you Auto Action Download users!
<r>I can make your text red<r>

pretty cool right?

Human... I remember your <b>Commit History<b>

ItsLJcool committed:
added "hx.ws" as an import (wont work neo but ok)

ItsLJcool committed:
fixed indentation

ItsLJcool committed:
Fixed some issues

ItsLJcool commented:
Uh not done witht his yet actually, lemme add smth rq

NexIsDumb commented:
i can already see the degenerates making buttplus addons
i love it

NexIsDumb commented:
BUTTPLUS😭

PetscopF commented:
The addon that adds an explosion to the screen as a separate window (using NDLLs) by secretly using a WebSocket to download the GIF and use it:

NexIsDumb commented:
codename streaming

PetscopF commented:
truely

TechnikTil commented:
aight who wanna help me with codename online

ItsLJcool commented:
> aight who wanna help me with codename online

Thats the entire reason why im adding this lol
I already made CNE Online but with this I can make it more efficent and better managed

PetscopF commented:
the seventeen player experience

ItsLJcool commented:
![image](https://github.com/user-attachments/assets/a018d98b-5c71-4381-a5a4-e68d9fbb17e4)
Committing code war crimes rn

PetscopF commented:
the concatenation experience (GONE WRONG)

TechnikTil commented:
> ![image](https://private-user-images.githubusercontent.com/54383469/400171231-a018d98b-5c71-4381-a5a4-e68d9fbb17e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwMzcwMzYsIm5iZiI6MTczNjAzNjczNiwicGF0aCI6Ii81NDM4MzQ2OS80MDAxNzEyMzEtYTAxOGQ5OGItNWM3MS00MzgxLWE1YTQtZTY4ZDlmYmIxN2U0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDAwMjUzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwZjIyYjEyZTk3ODRkOGFkNjZhMmE2MDhhYjI0ZWY1MGQ3NjQ5NzM2ZjYwMDk4NjVjNmFlZDU1N2I1OTkxN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AL6iNekrh4g_6ixgOqXXpns8Gbe66t9trz_4P22kHpQ) Committing code war crimes rn

what is wrong with u
i hope you burn /j

ItsLJcool commented:
> > ![image](https://private-user-images.githubusercontent.com/54383469/400171231-a018d98b-5c71-4381-a5a4-e68d9fbb17e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYwMzcwMzYsIm5iZiI6MTczNjAzNjczNiwicGF0aCI6Ii81NDM4MzQ2OS80MDAxNzEyMzEtYTAxOGQ5OGItNWM3MS00MzgxLWE1YTQtZTY4ZDlmYmIxN2U0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTA1VDAwMjUzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwZjIyYjEyZTk3ODRkOGFkNjZhMmE2MDhhYjI0ZWY1MGQ3NjQ5NzM2ZjYwMDk4NjVjNmFlZDU1N2I1OTkxN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AL6iNekrh4g_6ixgOqXXpns8Gbe66t9trz_4P22kHpQ) Committing code war crimes rn
> 
> what is wrong with u i hope you burn /j

![image](https://github.com/user-attachments/assets/4cf15634-f6b4-4ed6-b4b3-e127d177b471)
Updated :trollface:

ItsLJcool commented:
wait im so smart chat hol up

NeeEoo commented:
use an array please, its more optimized

TechnikTil commented:
> > aight who wanna help me with codename online
> 
> Thats the entire reason why im adding this lol I already made CNE Online but with this I can make it more efficent and better managed

can i help pleaseeeeeeeeee /hj

ItsLJcool commented:
> > > aight who wanna help me with codename online
> > 
> > 
> > Thats the entire reason why im adding this lol I already made CNE Online but with this I can make it more efficent and better managed
> 
> can i help pleaseeeeeeeeee /hj

What can you help with?

TechnikTil commented:
> > > > aight who wanna help me with codename online
> > > 
> > > 
> > > Thats the entire reason why im adding this lol I already made CNE Online but with this I can make it more efficent and better managed
> > 
> > 
> > can i help pleaseeeeeeeeee /hj
> 
> What can you help with?

i was joking, but kinda not cuz im just bored out of my mind rn

ItsLJcool commented:
Fair, I do want some ppl to help with like assets / concepts and playtesting

TechnikTil commented:
> Fair, I do want some ppl to help with like assets / concepts and playtesting

i can kinda draw, i can concept
and playtesting is playtesting lol

PetscopF commented:
> Fair, I do want some ppl to help with like assets / concepts and playtesting

I mean, I am able to playtest most of the time if necessary.

ItsLJcool commented:
> > Fair, I do want some ppl to help with like assets / concepts and playtesting
> 
> i can kinda draw, i can concept and playtesting is playtesting lol (also got a linux server running in my basement lmao)

Wait you do??
peak what would I pay a month?

NeeEoo commented:
i think 500$ per hour would be good for them

TechnikTil commented:
> > > Fair, I do want some ppl to help with like assets / concepts and playtesting
> > 
> > 
> > i can kinda draw, i can concept and playtesting is playtesting lol (also got a linux server running in my basement lmao)
> 
> Wait you do?? peak what would I pay a month?

$10/month max, i could settle for $5
just give me like 2 weeks to upgrade specs

ItsLJcool commented:
I mean I prob could pay but for now I can host the server for free on Pylex.

It's actually peak Node.js servers.

They recently made all free servers 24/7

TechnikTil commented:
probably do that, cuz my server rn just blatantly gives out my ip

ItsLJcool committed:
Added `WebSocketPacket` to be used with `WebSocketUtil`.

<b>Srt<b> was the person behind hanlding the data when we did Codename Engine Multiplayer Test.

ItsLJcool commented:
ok its ready to be merged now

ItsLJcool committed:
clarification

NexIsDumb commented:
lemme do some stuff

NexIsDumb committed:
allow scripting the entire raw websocket api

ItsLJcool committed:
reformatted the `WebSocketPacket` to use `Serializer` instead of a long string.

Also added Meta Data to the packet (can be disabled as a 3rd param)

ItsLJcool committed:
Merge branch 'internet-utils' of https://github.com/ItsLJcool/CodenameEngine into internet-utils

NexIsDumb committed:
moving to .system.net instead of .internet

so its all together with also yosh's socket
(important to remember that socket and websocket are two different things though)

ItsLJcool commented:
![image](https://github.com/user-attachments/assets/cb2972d6-64d4-426b-9093-ec87313bc166)

ItsLJcool committed:
"The => should only be in the not empty part" - Neo 2025

ItsLJcool committed:
Added Discord Meta Data (Username, Global name and Premium Type only)

ItsLJcool committed:
updated `WebSocketPacket` to actually you know do its job and not crash under cases

ItsLJcool committed:
erm i forgor about a trace line

TechnikTil commented:
cant wait for this to be merged

TechnikTil commented:
> Uh not done witht his yet actually, lemme add smth rq

btw you can turn the pr into a wip one if you didnt know

ItsLJcool commented:
> > Uh not done witht his yet actually, lemme add smth rq
> 
> btw you can turn the pr into a wip one if you didnt know

?

ItsLJcool commented:
> cant wait for this to be merged

Im making a server alongside developing the WebSocketUtil so I can push a JavaScript Template server for those who actually want to use it and stuff.

I should probably make it rn so I can add people to help tbh

ItsLJcool committed:
No longer sends Discord Metadata if your not connected.

ItsLJcool committed:
Merge branch 'CodenameCrew:main' into internet-utils

ItsLJcool committed:
Added `ServerPacketData`. Messages recieved by the WebSocket will either parse it as a `ServerPacketData` if it can, if not it will be a string.

Thanks to Neo for helping with that.

NexIsDumb commented:
Imma also make this as a draft till its done

ItsLJcool commented:
yeah couldn't figure out how to change it lmao

ItsLJcool committed:
`closeOnError` is default since you can handle your errors by default.

Basically the Server Template is done, and so I think WebSocket's are ready to be put into CNE.

ItsLJcool commented:
Server Template for the WebSocket:

https://github.com/ItsLJcool/WebSocket-Server-Template-for-CNE

ItsLJcool committed:
Merge branch 'CodenameCrew:main' into internet-utils

ItsLJcool committed:
Added Link in comments to Server Template

ItsLJcool committed:
Connecting to WebSocket is now Async. Can be disabled.

ItsLJcool committed:
forgor about stuff, also added Threaded Packet Sending (disabled by default)

ItsLJcool committed:
oop

NeeEoo commented:
Disable all async stuff for now, but keep the toggle

ItsLJcool committed:
I forgot the `onmessage` is being sent from a thread, so updated the `_threadedSend`

ItsLJcool committed:
MORE!!!

ItsLJcool committed:
testing packet storing

ItsLJcool commented:
> Disable all async stuff for now, but keep the toggle
👍

ItsLJcool committed:
might fix compiling issue?

NeeEoo commented on source/funkin/backend/system/net/WebSocketUtil.hx:
```diff
@@ -0,0 +1,259 @@
+	* Sends data to the server
+	**/
+	public function send(data) {	
+		var _data = null;
+		if (data is WebSocketPacket) _data = data.toString();
+		else _data = data;
```
Please give data and _data a type

ItsLJcool committed:
Merge remote-tracking branch 'upstream/main' into internet-utils

FuroYT commented:
> this KINDA has merge conflicts yk… rn at least… would be cool if u fixed (if you still wanna add this, that is)…

I think its unfinished iirc

ItsLJcool commented:
Deprechating this as its old asf

Planning on remaking it rn :trollface:

TechnikTil commented:
bruh

r6915ee commented:
bruh

Raltyro commented:
Ibwas going to lose hope
